src/core/Camera.js

```diff
@@ -679,7 +685,16 @@ Phaser.Camera.prototype = {
         if (vx <= this.bounds.x * this.scale.x)
         {
             this.atLimit.x = true;
-            this.view.x = this.bounds.x * this.scale.x;
+
+            // if smoothBounds enabled, interpolate to new view position (at rate this.lerp.x) instead of jumping.
+            if (this.smoothBounds)
+            {
+                this.view.x = this.game.math.linear(this.view.x, this.bounds.x * this.scale.x, this.lerp.x);
```

Comment:
The `checkBounds()` method shouldn't do the interpolation, as it could be called multiple times per game tick, I guess.